examples/plot_conditional_vs_marginal_xor_data.py

```diff
@@ -0,0 +1,159 @@
+#  Visualizing the decision function of the SVC
+# -----------------------------------------------------
+
+fig, ax = plt.subplots()
+ax.imshow(
+    Z.reshape(xx.shape),
+    interpolation="nearest",
+    extent=(xx.min(), xx.max(), yy.min(), yy.max()),
+    cmap="RdYlBu_r",
+    alpha=0.6,
+    origin="lower",
+    aspect="auto",
+)
+sns.scatterplot(
+    x=X[:, 0],
+    y=X[:, 1],
+    hue=Y,
+    ax=ax,
+    palette="muted",
+)
+ax.axis("off")
+ax.legend(
+    title="Class",
+)
+ax.set_title("Decision function of SVC with RBF kernel")
+plt.show()
+
+
+##############################################################################
+# The decision function of the SVC shows that the model is able to learn the
+# non-linear decision boundary of the XOR problem. It also highlights that knowing
+# the value of both features is necessary to classify each sample correctly.
+
+
+###############################################################################
+# Computing the conditional and marginal importance
+# -----------------------------------------------------
+# We first compute the marginal importance by fitting univariate models on each feature.
+# Then, we compute the conditional importance using the CPI class. The univarariate
+# models don't perform above chance, since solving the XOR problem requires to use both
+# features. Conditional importance, on the other hand, reveals that both features
+# are important (therefore rejecting the null hypothesis
+# :math:`Y \perp\!\!\!\perp X^1 | X^2`).
+cv = KFold(n_splits=5, shuffle=True, random_state=0)
+clf = SVC(kernel="rbf", random_state=0)
+# Compute marginal importance using univariate models
+marginal_scores = []
+for i in range(X.shape[1]):
+    feat_scores = []
+    for train_index, test_index in cv.split(X):
+        X_train, X_test = X[train_index], X[test_index]
+        y_train, y_test = Y[train_index], Y[test_index]
+
+        X_train_univariate = X_train[:, i].reshape(-1, 1)
+        X_test_univariate = X_test[:, i].reshape(-1, 1)
+
+        univariate_model = clone(clf)
+        univariate_model.fit(X_train_univariate, y_train)
+
+        feat_scores.append(univariate_model.score(X_test_univariate, y_test))
+    marginal_scores.append(feat_scores)
+
+###########################################################################
+
+importances = []
+for i, (train_index, test_index) in enumerate(cv.split(X)):
+    X_train, X_test = X[train_index], X[test_index]
+    y_train, y_test = Y[train_index], Y[test_index]
+
+    clf_c = clone(clf)
+    clf_c.fit(X_train, y_train)
+
+    vim = CPI(
+        estimator=clf_c,
+        method="decision_function",
+        loss=hinge_loss,
+        imputation_model_continuous=RidgeCV(np.logspace(-3, 3, 10)),
+        n_permutations=50,
+        random_state=0,
+    )
+    vim.fit(X_train, y_train)
+    importances.append(vim.importance(X_test, y_test)["importance"])
+
+importances = np.array(importances).T
+
```

Comment:
You should create a separation here and add a"numerical" presentation of the result.

Reply:
done